Parse/Internal/Utilities/FlexibleListWrapper.cs

```diff
@@ -28,11 +29,11 @@ class FlexibleListWrapper<TOut, TIn> : IList<TOut> {
     }
 
     public int IndexOf(TOut item) {
-      return toWrap.IndexOf((TIn)ParseClient.ConvertTo<TIn>(item));
+      return toWrap.IndexOf((TIn)Conversion.ConvertTo<TIn>(item));
```

Comment:
Again, important that the exception given here is useful. If we silently fail here we change behavior significantly, as `IndexOf(null)` is valid. Also, we literally cannot call `As` because we support value types here.